src/mlx90632.c

```diff
@@ -449,6 +449,11 @@ int32_t mlx90632_init(void)
     if (ret < 0)
         return ret;
 
+    if ((eeprom_version & 0x7F00) != MLX90632_XTD_RNG_KEY)
```

Comment:
Also move this check above read for `reg_status` to reduce discrepancy possibility.

Reply:
I see your point, but ERANGE is not really an error - it simply means that the device does not support extended range measurements. It can still be used for medical measurements. That is why I do this check after all other checks have passed - making sure that everything else is OK.

Reply:
If this is true, then we should not return negative value, but positive. Also that should be described in the comments in mlx90632.h